runtime/src/launch/slurm-srun/launch-slurm-srun.c

```diff
@@ -88,11 +88,21 @@ static sbatchVersion determineSlurmVersion(void) {
   }
 }
 
+static int nomultithread(int batch) {
+  char* hint = getenv("SLURM_HINT");
+  if (hint && !strcmp(hint, "nomultithread"))
+    return 1;
+  hint = getenv("SBATCH_HINT");
```

Comment:
This is functionally correct and granted we're not in a place where performance matters, but
```
if (batch) { hint = getenv("SBATCH_HINT"); if (hint && !strcmp(hint, ...
```
would still be prettier.

Reply:
`SLURM_HINT` works for srun and sbatch (and salloc) whereas `SBATCH_HINT` is sbatch specific.

Reply:
Oh, or did you mean just make that change for the second conditional?

So:

```c
static int nomultithread(int batch) {
  char* hint = getenv("SLURM_HINT");
  if (hint && !strcmp(hint, "nomultithread"))
    return 1;
  if (batch) {
    hint = getenv("SBATCH_HINT");
    if (hint && !strcmp(hint, "nomultithread"))
      return 1;
  }
  return 0;
}
```

Reply:
Right, don't bother looking at `SBATCH_HINT` unless we know we're doing batch.

Reply:
How bout:

```c
  char* hint;
  if ((hint = getenv("SLURM_HINT")) && !strcmp(hint, "nomultithread"))
    return 1;
  if (batch && (hint = getenv("SBATCH_HINT")) && !strcmp(hint, "nomultithread"))
    return 1;
  return 0;
```

Reply:
Updated to something closer to that (but using strcmp with `== 0` instead of `!` to match the rest of the file.)